enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberState.java

```diff
@@ -346,4 +378,23 @@ public abstract HighAvailabilityMemberState masterIsAvailable( HighAvailabilityM
     public abstract boolean isEligibleForElection();
 
     public abstract boolean isAccessAllowed();
+
+    private boolean transitionWasLegal = true;
```

Comment:
please move those fields to be before any methods, just after SLAVE enum value declaration finish